tutorials/first-steps-5.md

```diff
@@ -0,0 +1,126 @@
+<a href="https://raw.githubusercontent.com/skydive-project/skydive/master/etc/skydive.yml.default">here</a>. The config is stored in
+`/etc/skydive/skydive.yml`.
+
+For the agents, only the `analyzers` list is set to specify the analyzer address.
+
+{% highlight yaml %}
+analyzers:
+- 192.168.50.10:8082
+{% endhighlight %}
+
+For the analyzer there are more modifications. Elasticsearch is specified as backend for the flows :
+
+{% highlight yaml %}
+analyzer:
+  storage:
+    backend: elasticsearch
+{% endhighlight %}
+
+and the topology :
+
+{% highlight yaml %}
+graph:
+  backend: elasticsearch
+{% endhighlight %}
+
+The way the agents are connected together is also described in the config file. We are not going to detail here the syntax but basically it creates a `TOR1` node and two port nodes `TOR1_PORT1`, `TOR1_PORT2` connected to the `eth1` interface of the host agents.
+
+{% highlight yaml %}
+analyzer:
+  topology:
+    fabric:
+    - TOR1[Name=tor1] -> TOR1_PORT1[Name=port1, MTU=1500]
+    - TOR1_PORT1 -> *[Type=host, Name=agent1]/eth1
+    - TOR1[Name=tor1] -> TOR1_PORT2[Name=port2, MTU=1500]
+    - TOR1_PORT2 -> *[Type=host, Name=agent2]/eth1
+{% endhighlight %}
+
+<h2>Packet journey within a tunnel</h2>
+
+Now we have a properly working multi-node deployment, we can create a topology with tunnels. For that purpose we use Open vSwitch with GRE tunnels. The following command lines executed on the agent1 will create a network namespace connected to a bridge and will add a GRE interface :
+
+{% highlight shell %}
+vagrant ssh agent1
+
+sudo ip netns add ns1
+sudo ip link add ns1-eth0 type veth peer name eth0 netns ns1
+sudo ip link set ns1-eth0 up
+sudo ip netns exec ns1 ip link set eth0 up
+sudo ip netns exec ns1 ip address add 10.0.0.1/24 dev eth0
+sudo ovs-vsctl add-br br0
+sudo ovs-vsctl add-port br0 ns1-eth0
+sudo ovs-vsctl add-port br0 gre0 -- set interface gre0 type=gre options:remote_ip=192.168.50.30
+{% endhighlight %}
+
+Same thing on the second agent host with the following command lines :
+
+{% highlight shell %}
+vagrant ssh agent2
+
+sudo ip netns add ns2
+sudo ip link add ns2-eth0 type veth peer name eth0 netns ns2
+sudo ip link set ns2-eth0 up
+sudo ip netns exec ns2 ip link set eth0 up
+sudo ip netns exec ns2 ip address add 10.0.0.2/24 dev eth0
+sudo ovs-vsctl add-br br0
+sudo ovs-vsctl add-port br0 ns2-eth0
+sudo ovs-vsctl add-port br0 gre0 -- set interface gre0 type=gre options:remote_ip=192.168.50.20
+{% endhighlight %}
+
+<p>
+  Once executed, Skydive should have detected the two network namespaces with all the links and should look like this.
+</p>
+
+<p>
+  <img src="/assets/images/first-steps/multi-nodes-2.png"/>
+</p>
+
+Now we can create 3 captures, one on a the `eth1` interface of an agent host, and one on each Open vSwitch bridge. Having the capture created, we can use the packet injector to generate ping between the two network namespaces.
```

Comment:
one on the `eth1` interface of an agent host, and one other on each Open vSwitch bridge.
Once the captures are created, we can use the packet injector to generate pings between the two network namespaces.